[feat] 투표 옵션에 이미지, 코드블럭 추가

hwookim opened the pull request:
close #53 

## 💡 개요
<!-- 구현 내용 및 작업 했던 내역 -->
<!-- 작업 내용을 이미지나 gif로 첨부해도 좋습니다 -->
![스크린샷 2023-01-28 오후 1 58 31](https://user-images.githubusercontent.com/45786387/215242945-48ee1a71-de05-43c1-bf1f-d3a64bfee73c.png)
![스크린샷 2023-01-28 오후 1 58 45](https://user-images.githubusercontent.com/45786387/215242950-3b31041d-7807-4fa4-9c16-feca2604aaa2.png)

## 📝 작업 내용
<!-- 작업 내용 -->

- SelectOption에 이미지, 코드블럭 기능 추가
- CodeEditor 구현 <- 추후 생성 페이지에서도 이용할 예정

## ‼️ 주의 사항
<!-- 해당 작업에서 주의해아할 사항  -->

- 코드 블럭 표시 방식에 대해서는 논의 중인 내용이 있어서 이후 추가 작업이 필요할 것 같습니다!
- next에서 사용 가능하도록 설정 변경한 부분이 있음 [참고](https://github.com/uiwjs/react-textarea-code-editor/issues/31#issuecomment-909363339)
- 전체 document에 다크모드 활성화함


## 🔗 참고자료
<!-- 디자인 시안 링크 또는 레퍼런스 등 참고할만한 자료 -->

- https://github.com/uiwjs/react-textarea-code-editor

hwookim committed:
feat: DB 구조와 API docs를 참고한 공통 인터페이스 구현 및 적용

hwookim committed:
refactor: PostCommentRequest 구현

hwookim committed:
feat: api 스펙 변경에 맞춘 네이밍 변경

hwookim committed:
feat: SelectOption에 이미지 추가

- image mock data 추가
- 버튼으로 태그 변경

hwookim committed:
refactor: style -> styles

hwookim committed:
feat: CodeEditor 추가

hwookim committed:
wip

hwookim committed:
Merge remote-tracking branch 'origin/develop' into feat/053-topic-card-plus

# Conflicts:
#	__mocks__/data/voteOption.ts
#	src/components/common/TopicCard/SelectOption/SelectOption.tsx
#	src/components/common/TopicCard/TopicCard.stories.tsx
#	src/components/common/TopicCard/TopicCard.tsx
#	src/components/topic/TopicDetailMain/TopicDetailMain.tsx

hwookim committed:
feat: 코드블럭 추가

hwookim committed:
fix: 그라데이션 넓이 유동적으로 변경

KIMSEUNGGYU commented on __mocks__/data/voteOption.ts:
```diff
@@ -16,3 +18,66 @@ export const VOTE_OPTION2: VoteOption = {
+  codeBlock: {
+    language: 'js',
+    contents: `
+import React, { MouseEvent } from 'react';
+
+import { Languages } from '@src/components/common/CodeEditor';
+import Icon from '@src/components/common/Icon';
+import VoteOption from '@src/types/VoteOption';
+
```
해당 부분 다른데에서도 사용하는 거 같은데 mock_code 만들어서 처리하는건 어떤가요?

hwookim replied:
반영했습니당!

KIMSEUNGGYU commented on src/components/common/CodeEditor/CodeEditor.styles.tsx:
```diff
@@ -0,0 +1,8 @@
+import theme from '@src/styles/theme';
+
+export const CodeEditor = styled(TextAreaCodeEditor)`
+  ${theme.font.code}
```
스타일이 제대로 적용된건가요?? 속성이 정의가 되지 않고 값만 들어가 있는 거 같아요!

hwookim replied:
확인해보니 적용되고 있는 것 같은데 어떤 문제가 있을까요?
![image](https://user-images.githubusercontent.com/45786387/215279070-db84c383-2ddd-4a2a-85e8-1ac2d85a0cd3.png)

KIMSEUNGGYU replied:
![image](https://user-images.githubusercontent.com/45627868/215501847-a0f8a327-f14b-48a5-8ccf-ab9354fd2063.png)
1.  스토리북에서 확인해보니 기본 폰트 color 가 어두운거 같은데 정상 동작하시나요?? 
저희 배포된 스토리북(chromatic 및 local)으로 확인해도 해당 부분 컬러가 어둔운데 확인 부탁드립니다!
2. codeEditor 부분이 viewer 하고 edit 모두 사용하는지 궁금합니다.
   만약 viewer 로만 사용한다면 네이밍을 `CodeViwer`가 더 낫지 않을까 싶습니다!

KIMSEUNGGYU commented on src/components/common/CodeEditor/CodeEditor.tsx:
```diff
@@ -0,0 +1,17 @@
+
+export interface CodeEditorProps extends TextareaCodeEditorProps {
+  value: string;
+  language: Languages;
```
코드 에디터에 language 는 어떻게 사용하는건가요?
아직 제대로 적용이 안된건가요?

hwookim replied:
정상 적용되고 있는데 storybook에 제대로 된 인자를 안넣어서 혼란을 드린 것 같네요.
관련 storybook 업데이트 했습니다!

KIMSEUNGGYU commented on src/components/common/TopicCard/SelectOption/SelectOption.tsx:
```diff
@@ -13,22 +14,35 @@ interface SelectOptionProps extends VoteOption {
-      </S.Info>
+    <S.Container>
+      <S.SelectButton $result={result} $selected={selected} onClick={handleClick}>
+        {result && <S.ProgressBar className="progress" $rate={rate} $selected={selected} />}
```
해당 클래스컴포넌트에 progress 클래스 가 있는데 해당 부분 style 이 없는거 같은데 혹시 넣으신 의도가 궁금해요!

hwookim replied:
`SelectButton `내에서 `hover` 시 `.progress`의 `background-color`를 조정하는 로직이 있습니다!

KIMSEUNGGYU commented on src/components/common/TopicCard/TopicCard.tsx:
```diff
@@ -7,7 +7,7 @@ import VoteOption from '@src/types/VoteOption';
 import Icon from '../Icon';
 import SelectOption from './SelectOption';
-import * as S from './TopicCard.style';
+import * as S from './TopicCard.styles';
```
ㅋㅋㅋ 굿굿굿!! 👍👍👍😁

KIMSEUNGGYU reviewed (commented):
오오 점점 완성된 가는거 보니깐 좋네요!!
이번 UI 하면서 재미있었을거 같아요!!

hwookim committed:
refactor: codeblock mockdata화

hwookim committed:
test: CodeEditor에 대한 props 정의

hwookim reviewed (commented):
맞아요 만들면서 너무 재밌었던 것 같아요 ㅋㅋㅋ

리뷰에 답변 남기면서 몇가지는 반영했습니다!
확인 한 번 부탁드려요 👍🏼

KIMSEUNGGYU reviewed (approved):
CodeEditor 부분만 확인 부탁드립니다!
확인하시고 현우님 판단하에 처리하시면 괜찮을거 같아요!
나머지는 완벽합니다!! 👍👍👍

hwookim commented:
@KIMSEUNGGYU 
![image](https://user-images.githubusercontent.com/45786387/215524590-41e5b701-08e2-4236-a6ff-f088c40638c1.png)
저는 정상 동작하는 걸로 보이는데,,, 어떤 문제일까요 😭 
제 쪽에서 구현이 안되어서 가능하시다면 승규님께 확인 부탁드려도 괜찮을까요 ㅠㅠ

에디터에도 같이 쓸 생각으로 만든 거긴 합니다!
이 부분은 혜진님과도 논의가 한 번 더 필요할 것 같긴 하지만요.

hwookim committed:
fix: dark 모드 적용

hwookim committed:
Merge remote-tracking branch 'origin' into feat/053-topic-card-plus

hwookim committed:
feat: next에서 사용 가능하도록 변경

https://github.com/uiwjs/react-textarea-code-editor/issues/31#issuecomment-909363339

hwookim committed:
fix: 코드 블록에 스타일 제대로 안먹던 문제 해결

hwookim committed:
fix: 쓸데없는 tab 제거

KIMSEUNGGYU reviewed (approved):
현우님 CodeEditor 스타일 적용확인했습니다. 
근데 하나 더 이슈가 보입니다 ㅠㅠ

![image](https://user-images.githubusercontent.com/45627868/215745186-ad02441f-e271-45ce-9067-88447ad339d7.png)
드래그 했을때 작은 글씨로 덮어쓰이는 거 같은데! 한 번 확인 부탁드립니다!

hwookim committed:
fix: 폰트 변경이 정상적으로 작동하지 않던 문제 해결

hwookim commented:
후,,,, 되게 스타일이 이상하게 적용되고 있었네요...
에디터에서 textarea랑 실제 보여지는 text랑 따로 관리하다보니 폰트가 어그러져 있었습니다 ㅠㅠ
좀 더럽긴 하지만
```
export const CodeEditorWrapper = styled.div`
  ${theme.font.code}

  .w-tc-editor, * {
    ${theme.font.code}
  }
`;
```

이런 식으로 하위에 적용하니 되긴 하네요..!

KIMSEUNGGYU reviewed (approved):
현우님 완벽합니다!!
정말 수고하셨어요~~

여태 일하면서 작업하시다니 대단합니다!! 👍👍👍
